lib/Cucumber.j

```diff
@@ -103,98 +122,140 @@ function simulate_left_click_on_view(aKey, aValue, flags)
+{
+    simulate_left_click_on_view(aKey, aValue, flags);
+    simulate_left_click_on_view(aKey, aValue, flags);
+}
+
+function simulate_double_click_on_point(x, y, flags)
+{
+    simulate_left_click_on_point(x, y, flags);
+    simulate_left_click_on_point(x, y, flags);
+}
+
+function simulate_mouse_moved_on_view(aKey, aValue, flags)
 {
+    var objectID = _getObjectsWithKeyAndValue(aKey, aValue);
+
     if (!flags)
         flags = [];
 
-    [cucumber_instance simulateRightClick:[x, y, flags]];
+    [cucumber_instance simulateMouseMoved:[objectID, flags]];
 }
 
-function simulate_double_click_on_point(x, y, flags)
+function simulate_mouse_moved_on_point(x, y, flags)
 {
     if (!flags)
         flags = [];
 
-    [cucumber_instance simulateDoubleClick:[x, y, flags]];
+    [cucumber_instance simulateMouseMovedOnPoint:[x, y, flags]];
 }
 
 function simulate_dragged_click_view_to_view(aKey, aValue, aKey2, aValue2, flags)
 {
-    var objectID = _getObjectsWithKeyAndValue(aKey, aValue),
-        objectID2 = _getObjectsWithKeyAndValue(aKey2, aValue2);
+    simulate_mouse_moved_on_view(aKey, aValue, flags);
+    _simulate_mouse_down_on_view(aKey, aValue, flags, CPLeftMouseDown);
 
-    if (!flags)
-        flags = [];
-
-    [cucumber_instance simulateDraggedClickViewToView:[objectID, objectID2, flags]];
+    simulate_mouse_moved_on_view(aKey2, aValue2, flags);
+    _simulate_mouse_up_on_view(aKey2, aValue2, flags, CPLeftMouseUp);
 }
 
 function simulate_dragged_click_view_to_point(aKey, aValue, x, y, flags)
 {
+    simulate_mouse_moved_on_view(aKey, aValue, flags)
+    _simulate_mouse_down_on_view(aKey, aValue, flags, CPLeftMouseDown);
+
+    simulate_mouse_moved_on_point(x, y, flags);
+    _simulate_mouse_up_on_point(x, y, flags, CPLeftMouseUp);
+}
+
+function simulate_dragged_click_point_to_point(x, y, x2, y2, flags)
+{
+    simulate_mouse_moved_on_point(x, y, flags);
+    _simulate_mouse_down_on_point(x, y, flags, CPLeftMouseDown);
+
+    simulate_mouse_moved_on_point(x2, y2, flags);
+    _simulate_mouse_up_on_point(x2, y2, flags, CPLeftMouseUp);
+}
+
+function simulate_scroll_wheel_on_view(aKey, aValue, deltaX, deltaY, flags)
+{
     var objectID = _getObjectsWithKeyAndValue(aKey, aValue);
 
     if (!flags)
         flags = [];
 
-    [cucumber_instance simulateDraggedClickViewToView:[objectID, x, y, flags]];
+    [cucumber_instance simulateMouseMovedOnPoint:[aKey, aValue, deltaX, deltaY, flags]];
 }
 
-function simulate_dragged_click_point_to_point(x, y, x1, y2, flags)
+function _simulate_mouse_up_on_view(aKey, aValue, flags, mouseType)
 {
+    var objectID = _getObjectsWithKeyAndValue(aKey, aValue);
+
     if (!flags)
         flags = [];
 
-    [cucumber_instance simulateDraggedClickViewToView:[x, y, x2, y2, flags]];
+    [cucumber_instance simulateMouseUp:[objectID, flags, mouseType]];
 }
 
-function simulate_mouse_moved_on_point(x, y, flags)
+function _simulate_mouse_up_on_point(x, y, flags, mouseType)
```

Comment:
Not sure, it's a good way to debug things. People could use mouse_up and mouse_down to really understand what is going on with the app